Extension/src/LanguageServer/client.ts

```diff
@@ -2387,10 +2390,47 @@ export class DefaultClient implements Client {
         return this.requestWhenReady(() => this.languageClient.sendRequest(SwitchHeaderSourceRequest, params));
     }
 
+    private async updateActiveDocumentTextOptions(): Promise<void> {
+        const editor: vscode.TextEditor | undefined = vscode.window.activeTextEditor;
+        if (editor
+            && editor.document
+            && editor.document.uri.scheme === "file"
+            && (editor.document.languageId === "c"
+                || editor.document.languageId === "cpp"
+                || editor.document.languageId === "cuda-cpp")) {
+            // If using vcFormat, check for a ".editorconfig" file, and apply those text options to the active document.
+            const settings: CppSettings = new CppSettings();
```

Comment:
I believe you should be passing the rootUri here or else you might not get the settings for the current workspace folder.

Reply:
Hmm.  I'm wondering if this is correct in a few other places.  Should the existing occurrence in `onDidChangeSettings()` also use `this.RootUri`?

Reply:
It's incorrect whenever a "resource" or "machine-overridable" scope setting is accessed, which is the case for "formatting" in onDidChangeSettings -- we could have other cases too.